File: Chapter1/1.1IsUnique.py
#Determines if a string has all unique characters
def isUnique(string):
    characters = {}
    for c in string:
        if characters.get(c) == None:
            characters[c] = 1
        else:
            return False
    return True

string = input("Enter a string:")
print(isUnique(string))

File: Chapter1/1.1IsUnique.go
//Determines if a string has all unique characters
package main

import "fmt"

func isUnique(s string) bool{
	c := make(map[string]int)
	for _,i:= range(s) {
		if c[string(i)] == 0 {
			c[string(i)] = 1
		} else{
			return false
		}
	}
	return true
}

func main() {
	var str string
	fmt.Println("Enter a string:")
	fmt.Scan(&str)
	fmt.Println(isUnique(str))
}